astronomer/providers/amazon/aws/example_dags/example_emr_eks_containers_job.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import os
+from datetime import datetime, timedelta
+
+import boto3
+from airflow import DAG
+from airflow.operators.bash import BashOperator
+from airflow.operators.python import PythonOperator
+from airflow.providers.amazon.aws.operators.emr import EmrContainerOperator
+from botocore.exceptions import ClientError
+
+from astronomer.providers.amazon.aws.sensors.emr import EmrContainerSensorAsync
+
+# [START howto_operator_emr_eks_env_variables]
+VIRTUAL_CLUSTER_ID = os.getenv("VIRTUAL_CLUSTER_ID", "xxxxxxxx")
+AWS_CONN_ID = os.getenv("ASTRO_AWS_CONN_ID", "aws_default")
+JOB_ROLE_ARN = os.getenv("JOB_ROLE_ARN", "arn:aws:iam::121212121212:role/test_iam_job_execution_role")
+# [END howto_operator_emr_eks_env_variables]
+
+# Job role name and policy name attached to the role
+JOB_EXECUTION_ROLE = os.getenv("JOB_EXECUTION_ROLE", "test_iam_job_execution_role")
+DEBUGGING_MONITORING_POLICY = os.getenv("DEBUGGING_MONITORING_POLICY", "test_debugging_monitoring_policy")
+CONTAINER_SUBMIT_JOB_POLICY = os.getenv(
+    "CONTAINER_SUBMIT_JOB_POLICY", "test_emr_container_submit_jobs_policy"
+)
+JOB_EXECUTION_POLICY = os.getenv("JOB_EXECUTION_POLICY", "test_job_execution_policy")
+MANAGE_VIRTUAL_CLUSTERS = os.getenv("MANAGE_VIRTUAL_CLUSTERS", "test_manage_virtual_clusters")
+
+EKS_CONTAINER_PROVIDER_CLUSTER_NAME = os.getenv(
+    "EKS_CONTAINER_PROVIDER_CLUSTER_NAME", "providers-team-eks-cluster"
+)
+KUBECTL_CLUSTER_NAME = os.getenv("KUBECTL_CLUSTER_NAME", "providers-team-eks-namespace")
+VIRTUAL_CLUSTER_NAME = os.getenv("EMR_VIRTUAL_CLUSTER_NAME", "providers-team-virtual-eks-cluster")
+AWS_ACCESS_KEY_ID = os.getenv("AWS_ACCESS_KEY_ID", "xxxxxxx")
+AWS_SECRET_ACCESS_KEY = os.getenv("AWS_SECRET_ACCESS_KEY", "xxxxxxxx")
+AWS_DEFAULT_REGION = os.getenv("AWS_DEFAULT_REGION", "us-east-2")
+
+default_args = {
+    "execution_timeout": timedelta(minutes=30),
+}
+
+
+def create_emr_virtual_cluster_func():
+    """Create EMR virtual cluster in container"""
+    client = boto3.client("emr-containers")
+    try:
+        response = client.create_virtual_cluster(
+            name=VIRTUAL_CLUSTER_NAME,
+            containerProvider={
+                "id": EKS_CONTAINER_PROVIDER_CLUSTER_NAME,
+                "type": "EKS",
+                "info": {"eksInfo": {"namespace": KUBECTL_CLUSTER_NAME}},
+            },
+        )
+        os.environ["VIRTUAL_CLUSTER_ID"] = response["id"]
```

Comment:
Can we push this xcom instead of env because this env will be machine specific

Reply:
In sync version example VIRTUAL_CLUSTER_ID is used as the env variables, to make it as self sufficient DAG we are setting it up as env variable. we make changes here we need to make it in sync version also